### PR TITLE
Closes #1572 Consolidate new environment objects into admiral_environment

### LINKED ISSUE #1572: Feature Request: Create only a single environment for admiral in a separate file
### Feature Idea

Right now environments are created in multiple files and are not documented.

### Relevant Input

_No response_

### Relevant Output

_No response_

### Reproducible Example/Pseudo Code

_No response_

### PR DESCRIPTION
**Pull request by zdz2101**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/main/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/main/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/main/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by bms63 — COMMENTED** [review]
Did we create any guidance around this in our developer vignettes?  This is all very new to me and I realize we have been using this method for a little while, but was unaware and makes me a little nervous.

**Review by bundfussr — APPROVED** [review]

**Review by bundfussr — APPROVED** [review]

**Review by bms63 — CHANGES_REQUESTED** [review]
Does this warrant a test that a new environment can be created?  I feel like this could be tested???  Or do the affected functions already have test coverages for new environments??

**Comment by zdz2101** [comment]
> Does this warrant a test that a new environment can be created?  I feel like this could be tested???  Or do the affected functions already have test coverages for new environments??

The affected functions' tests are already using this new consolidated environment object as well.

File: R/admiral_environment.R [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# admiral_options.R ----
+## set_admiral_options
+admiral_environment$admiral_options <- list(
+  # future_input = vars(...), nolint
+  subject_keys = vars(STUDYID, USUBJID)
+)
+
+######################################################
+### To enhance feature and add inputs as necessary ###
+######################################################
```
Comment (bms63) [inline R/admiral_environment.R:12]:
```suggestion
# To enhance features and add inputs as necessary 
```

File: R/admiral_environment.R [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+# 1. Add additional options such as future_input as shown commented above
```
Comment (bms63) [inline R/admiral_environment.R:14]:
```suggestion
# 1. Add additional options such as future_input as shown in comment above
```

File: R/admiral_environment.R [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+# 2. Update @params with future_input in set_admiral_options roxygen documentation
+# 3. Add future_input into set_admiral_options() formals and body
+
+# derive_merged.R ----
+## derive_vars_merged_lookup
+admiral_environment$nmap <- NULL
+
+# duplicates.R ----
+## signal_duplicate_records
+admiral_environment$duplicates <- NULL
```
Comment (bms63) [inline R/admiral_environment.R:24]: Should we say something like See `... .R`  for usage

**Review by bms63 — APPROVED** [review]
So other musings...should this have a Roxygen header.  I'm not sure if it can be displayed on our reference documentation page, but it would be nice to showcase and point to this functionality a little more???  No problem if we jsut want to bury it. :)

**Comment by bundfussr** [comment]
I think for developers and contributors it would be nice to have an overview of the objects in `admiral_environment` in the reference. But it should be clear that it is not of interest for users as they should never change or access these objects directly.

Most likely this requires a bit more of thought. There are more functions which are of interest for developers only, e.g, `assert_db_requirements()` or `extend_source_datasets()`. So maybe we could add a "Functions for developers and contributors" section in the next release (I think it's a bit late for this release).

**Comment by bms63** [comment]
@zdz2101 Can you add a roxygen header outlining what are goals are here with the enviroment.R file so it has a more standard feel to the file.  Use the `@nord` tag so it is not rendered onto our reference page.  This will be nice for developers and for developers who need a refresher on what this file is doing (like me).

**Review by bms63 — APPROVED** [review]
LGTM

**Review by bms63 — APPROVED** [review]
Just checking the Templates

**Pull request merged**